aptos-move/framework/supra-framework/sources/pbo_delegation_pool.move

```diff
@@ -7869,12 +7870,184 @@ module supra_framework::pbo_delegation_pool {
+        let delegator_address_vec = vector[delegator_address, @0x020];
+        let principle_stake = vector[100 * ONE_SUPRA, 200 * ONE_SUPRA];
+        let coin = stake::mint_coins(300 * ONE_SUPRA);
+        let principle_lockup_time = 7776000; // 3 month cliff
+        let multisig = generate_multisig_account(validator, vector[@0x12134], 2);
+
+        initialize_test_validator(
+            validator,
+            0,
+            true,
+            true,
+            0,
+            delegator_address_vec,
+            principle_stake,
+            coin,
+            option::some(multisig),
+            vector[1],
+            10,
+            principle_lockup_time,
+            LOCKUP_CYCLE_SECONDS // monthly unlocking
+        );
+        let funder_signer = account::create_signer_for_test(multisig);
+        let funder = signer::address_of(&funder_signer);
+        let validator_address = signer::address_of(validator);
+        let pool_address = get_owned_pool_address(validator_address);
+        let one_year_in_secs = 31536000;
+        let reward_period_start_time_in_sec = timestamp::now_seconds();
+        staking_config::initialize_rewards(
+            supra_framework,
+            fixed_point64::create_from_rational(1, 100),
+            fixed_point64::create_from_rational(1, 100),
+            one_year_in_secs,
+            reward_period_start_time_in_sec,
+            fixed_point64::create_from_rational(0, 100)
+        );
+
+        let new_delegator_address = @0x0215;
+        let new_delegator_address2 = @0x0216;
+
+        stake::mint(&funder_signer, 200 * ONE_SUPRA);
+        assert!(
+            coin::balance<SupraCoin>(funder) == (200 * ONE_SUPRA),
+            0
+        );
+
+        fund_delegators_with_stake(
+            &funder_signer,
+            pool_address,
+            vector[new_delegator_address, new_delegator_address2],
+            vector[30 * ONE_SUPRA, 70 * ONE_SUPRA]
+        );
+
+        fund_delegators_with_locked_stake(
+            &funder_signer,
+            pool_address,
+            vector[new_delegator_address, new_delegator_address2],
+            vector[30 * ONE_SUPRA, 70 * ONE_SUPRA]
+        );
+        assert!(coin::balance<SupraCoin>(funder) == 0, 0);
+
+        // 3 month
+        timestamp::fast_forward_seconds(LOCKUP_CYCLE_SECONDS);
+        end_aptos_epoch();
+        timestamp::fast_forward_seconds(LOCKUP_CYCLE_SECONDS);
+        end_aptos_epoch();
+        timestamp::fast_forward_seconds(LOCKUP_CYCLE_SECONDS);
+        end_aptos_epoch();
+
+        let new_delegator_address_signer = account::create_signer_for_test(new_delegator_address);
+        unlock(&new_delegator_address_signer, pool_address, 30 * ONE_SUPRA);
+        unlock(&new_delegator_address_signer, pool_address, 1 * ONE_SUPRA);
+    }
+
+    #[
+    test(
+        supra_framework = @supra_framework,
+        validator = @0x123,
+        delegator = @0x010,
+    )
+    ]
+    #[expected_failure(abort_code = 65561)]
```

Comment:
where you write `expected_failure` test, you need to define `location` with abort_code, just to remove compiler warning.